esphome/components/ld24xx/ld24xx.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "esphome/core/defines.h"
+
+#include <memory>
+
+#ifdef USE_SENSOR
+#include "esphome/core/helpers.h"
+#include "esphome/components/sensor/sensor.h"
+
+#define SUB_SENSOR_WITH_DEDUP(name, dedup_type) \
+ protected: \
+  ld24xx::SensorWithDedup<dedup_type> *name##_sensor_{nullptr}; \
+\
+ public: \
+  void set_##name##_sensor(sensor::Sensor *sensor) { \
+    this->name##_sensor_ = new ld24xx::SensorWithDedup<dedup_type>(sensor); \
+  }
+#endif
+
+#define highbyte(val) (uint8_t)((val) >> 8)
+#define lowbyte(val) (uint8_t)((val) &0xff)
+
+namespace esphome {
+namespace ld24xx {
+
+#ifdef USE_SENSOR
+// Helper class to store a sensor with a deduplicator & publish state only when the value changes
+template<typename T> class SensorWithDedup {
+ public:
+  SensorWithDedup(sensor::Sensor *sens = nullptr) : sens(sens) {}
```

Comment:
I don't see how this would ever be called with nullptr. If it is, then its a programming error and should crash :stuck_out_tongue_closed_eyes: 
```suggestion
  SensorWithDedup(sensor::Sensor *sens) : sens(sens) {}
```